lib/puppet/resource/type.rb

```diff
@@ -64,49 +59,6 @@ class Puppet::Resource::Type
     define_method("#{t}?") { self.type == t }
   end
 
-  require 'puppet/indirector'
-  extend Puppet::Indirector
-  indirects :resource_type, :terminus_class => :parser
-
-  def self.from_data_hash(data)
```

Comment:
Right, you get `Puppet::Network::FormatSupport` from extending `Puppet::Indirector` so yeah strictly speaking I don't think we'll be able to call these methods anymore for serialization because we've stopped indirecting. My fear though is that they're being used outside of that for some other serialization, possibly externally? We never deprecated from_/to_data_hash for `puppet/resource/type` and they're not marked private. Maybe I'm being too cautious but it makes me nervous:)